### PR TITLE
Change landing page to be `profiling-results`

### PR DESCRIPTION
**Pull request by willGraham01**
For https://github.com/UCL/TLOmodel/issues/1355

- Changes the index page of the profiling site to redirect to the `profiling-results` page by default.

Note that the current `index.rst` contents have been moved to `contents.rst` so that the `sphinx-build` still works. This change just means that the `index.html` page immediately sends a person to the profiling results.

- The profiling results table is now ordered by newest first.
- Commit SHAs in the profiling results table can be clicked to be taken to the corresponding commit in the main TLO repo.

#### Flag any other aesthetic changes that we need to make here in PR discussion.

### CODE REVIEW

**Review by matt-graham — COMMENTED** [review]
Do we need to have the redirect here? Could we not just keep the current index page but include the table of results within that page? 

In the docs on the main `TLOmodel` repository we generally follow the pattern of having any machine generated reStructuredText files having underscored prefixed names (and not being version controlled) with these files then included in some other reStructuredText file using the `.. include::` directive, for example [here](https://github.com/UCL/TLOmodel/blob/67c80589a2e0f274ac12c87cd76876df3010d51d/docs/data_sources.rst?plain=1#L9C1-L9C13) and [here](https://github.com/UCL/TLOmodel/blob/67c80589a2e0f274ac12c87cd76876df3010d51d/docs/hsi_events.rst?plain=1#L17).

So here we could have the website builder create a reStructuredText file with the table of profiling results and then just directly include this on the landing page? That would also save needing to have something like the placeholder

https://github.com/UCL/TLOmodel-profiling/blob/2535dcba351db4f4bb8503a83e08c9f0a2d42612/source/profiling.rst?plain=1#L6

we have currently (as we just output to a predetermined file name and include that file rather than replacing the placeholder).

On the other hand, there is nothing specifically wrong with the current redirect solution that I can see, it just seems a bit more complex than necessary (and I guess might be somewhat device / browser dependent on whether the redirect works or not).

**Comment by tamuri** [comment]
Yes, the redirect feels clunky.

**Comment by willGraham01** [comment]
My reasoning for the redirect was that it's the least noisy solution for what we initially wanted to achieve (since it saved going through the rst files and updating links), but yeah now I think about it I could've just moved the profiling results to that page :sweat_smile: 

Have also changed the `profiling.rst` (and `run-statistics.rst`) files to have the lookup table / plots `.. included:`ed from files with underscores like in the main TLO repo, rather than using the placeholders.

File: source/_profiling_table.rst
Comment (matt-graham) [inline source/_profiling_table.rst:1]: Can this file not be removed now? Deleting locally the build still seems to work without any issues.

File: source/conf.py
Comment (matt-graham) [inline source/conf.py:1]: Changing line 46 to
```
exclude_patterns = ["_*.rst"]
```
will prevent orphaned HTML files being generated for the `_profiling_table.rst` and `_stats_plots.rst` files that are just used as includes for other reStructuredText files (we actually don't do this currently on main `TLOmodel` docs as for example https://www.tlomodel.org/_data_sources.html exists but we probably should!).

**Review by matt-graham — CHANGES_REQUESTED** [review]
Thanks @willGraham01 for the updates. Tested and this works locally and looks generally good to me, other than a couple of minor suggestions / comments.

**Review by matt-graham — APPROVED** [review]

**Pull request merged**